runtime/jcl/common/java_lang_Class.cpp

```diff
@@ -1812,17 +1812,6 @@ Java_java_lang_Class_getNestHostImpl(JNIEnv *env, jobject recv)
 
 	if (NULL == nestHost) {
 		J9Class *clazzToUse = clazz;
-#if JAVA_SPEC_VERSION >= 15
```

Comment:
Why is this code being removed?

Reply:
Our previous implementation used Unsafe.defineAnonymousClass() to define hidden classes. It did not take ClassOption.NESTMATE that is passed in, so we returned the nest host of hostClass for anonymous classes here as a temporary workaround to make java -version working. Change in this PR is taking care of the ClassOption.NESTMATE and setting J9AccHiddenClassOptionNestmate. The change in loadAndVerifyNestHost() in this PR is now checking J9AccHiddenClassOptionNestmate and returning the correct nest host for hidden classes, so this piece of code can be removed.